homeassistant/components/openalpr.py

```diff
@@ -0,0 +1,395 @@
+"""
+Component that will help set the openalpr for video streams.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/openalpr/
+"""
+from base64 import b64encode
+import logging
+import os
+from time import time
+
+import requests
+import voluptuous as vol
+
+from homeassistant.config import load_yaml_config_file
+from homeassistant.const import (
+    CONF_API_KEY, CONF_NAME, CONF_USERNAME, CONF_PASSWORD, ATTR_ENTITY_ID,
+    EVENT_HOMEASSISTANT_STOP)
+from homeassistant.components.ffmpeg import (
+    get_binary, run_test, CONF_INPUT, CONF_EXTRA_ARGUMENTS)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_component import EntityComponent
+
+DOMAIN = 'openalpr'
+DEPENDENCIES = ['ffmpeg']
+REQUIREMENTS = [
+    'git+https://github.com/pvizeli/cloudapi.git'
+    '#egg=openalpr_api&subdirectory=python']
+
+_LOGGER = logging.getLogger(__name__)
+
+SERVICE_SCAN = 'scan'
+SERVICE_RESTART = 'restart'
+
+EVENT_FOUND = 'openalpr.found'
+
+ATTR_PLATE = 'plate'
+
+
+ENGINE_LOCAL = 'local'
+ENGINE_CLOUD = 'cloud'
+
+RENDER_IMAGE = 'image'
+RENDER_FFMPEG = 'ffmpeg'
+
+OPENALPR_REGIONS = [
+    'us',
+    'eu',
+    'au',
+    'auwide',
+    'gb',
+    'kr',
+    'mx',
+    'sg',
+]
+
+CONF_RENDER = 'render'
+CONF_ENGINE = 'engine'
+CONF_REGION = 'region'
+CONF_RUNTIME = 'runtime'
+CONF_INTERVAL = 'interval'
+CONF_ENTITIES = 'entities'
+
+DEFAULT_NAME = 'OpenAlpr'
+DEFAULT_ENGINE = ENGINE_LOCAL
+DEFAULT_RENDER = RENDER_FFMPEG
+DEFAULT_INTERVAL = 2
+
+DEVICE_SCHEMA = vol.Schema({
+    vol.Required(CONF_INPUT): cv.string,
+    vol.Optional(CONF_INTERVAL, default=DEFAULT_INTERVAL): cv.positive_int,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_RENDER, default=DEFAULT_RENDER):
+        vol.In([RENDER_IMAGE, RENDER_FFMPEG]),
```

Comment:
Yeah, FFMPEG is only needed if you use a video stream. I've add that to description. Ha-FFMPEG library is written in pure python and very lightway, so it should not be bad is it not used.